src/pages/inventory/index.tsx

```diff
@@ -0,0 +1,265 @@
+  const [currentPage, setCurrentPage] = useState(1);
+  const itemsPerPage = 10;
+
+  const indexOfLastItem = currentPage * itemsPerPage;
+  const indexOfFirstItem = indexOfLastItem - itemsPerPage;
+  const currentItems = data.slice(indexOfFirstItem, indexOfLastItem);
+
+  const handleTypeClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorType(event.currentTarget);
+  };
+  const handleCategoryClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorCategory(event.currentTarget);
+  };
+  const handleStatusClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorStatus(event.currentTarget);
+  };
+
+  const handleTypeClose = () => {
+    setAnchorType(null);
+  }
+
+  const handleMenuTypeClick = (value: string) => {
+    setType(value)
+    handleFilter();
+    handleTypeClose();
+  }
+
+  const handleCategoryClose = () => {
+    setAnchorCategory(null);
+  }
+
+  const handleMenuCategoryClick = (value: string) => {
+    setCategory(value);
+    handleFilter();
+    handleCategoryClose();
+  }
+
+  const handleStatusClose = () => {
+    setAnchorStatus(null);
+  }
+
+  const handleMenuStatusClick = (value: string) => {
+    setStatus(value);
+    handleFilter();
+    handleStatusClose();
+  }
+
+  const clearTypeFilter = () => {
+    setType('');
+  }
+  const clearCategoryFilter = () => {
+    setCategory('');
+  }
+  const clearStatusFilter = () => {
+    setStatus('');
+  }
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearch(event.target.value)
+  }
+
+  const handlePageChange = (_event: React.ChangeEvent<unknown>, value: number) => {
+    setCurrentPage(value);
+  }
+
+  const handleFilter = () => {
+    const searchFiltered = originalData.filter((row: { item: string, type: string, category: string, quantity: number; }) => {
+
+      const matchesType = type ? row.type.toLowerCase().includes(type.toLowerCase()) : true;
+
+      const matchesCategory = category ? row.category.toLowerCase().includes(category.toLowerCase()) : true;
+
+      // const matchesStatus = status ? item.status.toLowerCase().includes(status.toLowerCase()) : true; //
+
+      const matchesSearch = search
+        ? row.item.toLowerCase().includes(search.toLowerCase()) ||
+        row.type.toLowerCase().includes(search.toLowerCase()) ||
+        row.category.toLowerCase().includes(search.toLowerCase()) ||
+        // item.status.toLowerCase().includes(search.toLowerCase()) ||
+        row.quantity.toString().toLowerCase().includes(search.toLowerCase())
+        : true;
+
+      return matchesType && matchesCategory && matchesSearch;
+    });
+
+    setData(searchFiltered);
+  };
+
+  const supabase = createClient(import.meta.env.VITE_SUPABASE_URL, import.meta.env.VITE_SUPABASE_ANON_KEY); // When we deploy this application or when we create a server, this will have to be changed to server side so that the anon_key can be hidden.
```

Comment:
Good catch. However, during build and deploy the .env file is populated and then included in the firebase package. See this:  

Since Firebase is using a static server, environment variables are not set on the server.  The .env is updated during deployment to provide secrecy regarding account keys.  See https://github.com/digitalaidseattle/das-admin-template/blob/main/.github/workflows/firebase-hosting-merge.yml for an example.

Reply:
Could you help me understand about the static server? Based on what you're saying, my interpretation is that because firebase is using a static server, it's building everything on the server side and serving just the HTML to the browser. Therefore, the .env will not be exposed? 

I thought because the client-side is making the API call, it's grabbing the keys from the .env file

Reply:
Great question. Here is chatgpt's answer, better than mine. It is the last question I asked in this thread: https://chatgpt.com/share/66ec7b04-44dc-800e-a983-e0a46f41f85a